go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -89,7 +89,9 @@ type serviceConfig struct {
 //
 // `add` parameter indicates if the flows should exist or be removed from the cache
 // `hasLocalHostNetworkEp` indicates if at least one host networked endpoint exists for this service which is local to this node.
-func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add, hasLocalHostNetworkEp bool) {
+// `addressFamily` parameter limits the scope of the update to the given addressFamily (0 for both families, 4 for IPv4 only,
+// 6 for IPv6 only).
+func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add, hasLocalHostNetworkEp bool, addressFamily int) {
```

Comment:
IMHO, this change is too much for the once-in-lifetime dualstack conversion. I think accepting it as slightly disruptive and then proceeding with retries in the test or a slight sleep until rules get created make more sense. @trozet wdyt?
I'm a bit skeptical in this approach of keeping ipv4 flows as is and recreating ipv6 flows with all these "ifs"